packages/apple-pay-integration/src/apple-pay-customer-strategy.ts

```diff
@@ -543,29 +543,39 @@ export default class ApplePayCustomerStrategy implements CustomerStrategy {
     }
 
     private async _getBraintreeDeviceData() {
-        const data = await this._braintreeIntegrationService.getDataCollector();
+        const braintreePaymentMethod = this._paymentIntegrationService
+            .getState()
+            .getPaymentMethod(ApplePayGatewayType.BRAINTREE);
 
-        return data.deviceData;
+        if (braintreePaymentMethod?.clientToken) {
+            const data = await this._braintreeIntegrationService.getDataCollector();
+
+            return data.deviceData;
+        }
     }
 
     private async _initializeBraintreeIntegrationService() {
-        const state = await this._paymentIntegrationService.loadPaymentMethod(
-            ApplePayGatewayType.BRAINTREE,
-        );
+        try {
+            await this._paymentIntegrationService.loadPaymentMethod(ApplePayGatewayType.BRAINTREE);
 
-        const storeConfig = state.getStoreConfigOrThrow();
+            const state = this._paymentIntegrationService.getState();
 
-        const braintreePaymentMethod: PaymentMethod = state.getPaymentMethodOrThrow(
-            ApplePayGatewayType.BRAINTREE,
-        );
+            const storeConfig = state.getStoreConfigOrThrow();
 
-        if (!braintreePaymentMethod.clientToken || !braintreePaymentMethod.initializationData) {
-            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
-        }
+            const braintreePaymentMethod: PaymentMethod = state.getPaymentMethodOrThrow(
+                ApplePayGatewayType.BRAINTREE,
+            );
 
-        this._braintreeIntegrationService.initialize(
-            braintreePaymentMethod.clientToken,
-            storeConfig,
-        );
+            if (!braintreePaymentMethod.clientToken || !braintreePaymentMethod.initializationData) {
+                throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+            }
+
+            this._braintreeIntegrationService.initialize(
+                braintreePaymentMethod.clientToken,
+                storeConfig,
+            );
+        } catch (_) {
+            // we don't need to do anything in this block
```

Comment:
I see `// we don't need to do anything in this block` appears twice in the PR.
Could you please explain what would happen if an error occurs in the middle?

Reply:
@bc-peng This will mean that the braintree payment method cannot be initialized because we have enabled the appropriate feature in the CP